src/environmentd/tests/server.rs

```diff
@@ -677,3 +677,106 @@ fn test_storage_usage_collection_interval() -> Result<(), Box<dyn Error>> {
+        let server = util::start_server(config.clone())?;
+        let mut client = server.connect(postgres::NoTls)?;
+        // Retry because it may take some time for the initial snapshot to be taken.
+        Retry::default().max_duration(Duration::from_secs(5)).retry(|_| {
+            Ok::<f64, String>(
+                client
+                    .query_one(
+                        "SELECT EXTRACT(EPOCH FROM MAX(collection_timestamp))::float8 FROM mz_storage_usage;",
+                        &[],
+                    )
+                    .map_err(|e| e.to_string())?
+                    .try_get::<_, f64>(0)
+                    .map_err(|e| e.to_string())?,
+            )
+        })?
+    };
+
+    std::thread::sleep(storage_usage_collection_interval);
+
+    // Another storage usage collection should be scheduled immediately.
+    {
+        let server = util::start_server(config)?;
+        let mut client = server.connect(postgres::NoTls)?;
+
+        // Retry until storage usage is updated.
+        Retry::default().max_duration(Duration::from_secs(2)).retry(|_| {
+            let updated_timestamp = client
+                .query_one("SELECT EXTRACT(EPOCH FROM MAX(collection_timestamp))::float8 FROM mz_storage_usage;", &[])
+                .map_err(|e| e.to_string())?
+                .try_get::<_, f64>(0)
+                .map_err(|e| e.to_string())?;
+
+            if updated_timestamp > initial_timestamp {
+                Ok(())
+            } else {
+                Err(format!("updated storage collection timestamp {updated_timestamp} is not greater than initial timestamp {initial_timestamp}"))
+            }
+        })?;
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_storage_usage_doesnt_update_between_restarts() -> Result<(), Box<dyn Error>> {
+    mz_ore::test::init_logging();
+
+    let data_dir = tempfile::tempdir()?;
+    let storage_usage_collection_interval = Duration::from_secs(3);
+    let config = util::Config::default()
+        .with_storage_usage_collection_interval(storage_usage_collection_interval)
+        .data_directory(data_dir.path());
+
+    // Wait for initial storage usage collection.
+    let initial_timestamp: f64 = {
+        let server = util::start_server(config.clone())?;
+        let mut client = server.connect(postgres::NoTls)?;
+        // Retry because it may take some time for the initial snapshot to be taken.
+        Retry::default().max_duration(Duration::from_secs(5)).retry(|_| {
+            Ok::<f64, String>(
+                client
+                    .query_one(
+                        "SELECT EXTRACT(EPOCH FROM MAX(collection_timestamp))::float8 FROM mz_storage_usage;",
+                        &[],
+                    )
+                    .map_err(|e| e.to_string())?
+                    .try_get::<_, f64>(0)
+                    .map_err(|e| e.to_string())?,
+            )
+        })?
+    };
+
+    std::thread::sleep(Duration::from_secs(2));
```

Comment:
I'm also worried about the 2 here racing with the 3 second collection interval above and causing occasional flakes. This test looks like we never expect it to make a second refresh. Can the collection interval be bumped to something much higher so that if the check below gets delayed by a bit it'll still almost certainly succeed?

Reply:
Yep, just bumped it to 60 seconds